sdcm/utils/nemesis_utils/indexes.py

```diff
@@ -0,0 +1,98 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2023 ScyllaDB
+
+import logging
+import random
+import time
+from typing import Tuple
+
+from sdcm.cluster import BaseNode
+from sdcm.sct_events import Severity
+from sdcm.sct_events.database import DatabaseLogEvent
+from sdcm.sct_events.filters import DbEventsFilter
+from sdcm.sct_events.system import InfoEvent
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_table_with_biggest_partitions_count(session) -> Tuple[str, str]:
+    """Gets table with the most number of partitions"""
+    LOGGER.debug('Getting table with the most number of partitions')
+    res = session.execute('select keyspace_name, table_name, partitions_count from system.size_estimates')
+    rows_sorted = sorted(res, key=lambda x: x.partitions_count, reverse=True)
+    ks, cf = rows_sorted[0].keyspace_name, rows_sorted[0].table_name
+    LOGGER.debug('Table with the most number of partitions is {}.{}: {} partitions'.format(
+        ks, cf, rows_sorted[0].partitions_count))
+    return ks, cf
+
+
+def get_random_column_name(session, ks, cf) -> str:
+    res = session.execute(f"SELECT column_name FROM system_schema.columns"
+                          f" WHERE keyspace_name = '{ks}'"
+                          f" AND table_name = '{cf}'"
+                          f" AND kind in ('static', 'regular')"
+                          f" ALLOW FILTERING")
+    column = random.choice(list(res)).column_name
+    return column
+
+
+def create_index(session, ks, cf, column) -> str:
+    LOGGER.info('starting creating index: {}.{}("{}")'.format(ks, cf, column))
+    index_name = f"{cf}_{column}_nemesis".lower()
+    session.execute(f'CREATE INDEX {index_name} ON {ks}.{cf}("{column}")')
+    return index_name
+
+
+def wait_for_index_to_be_built(node: BaseNode, ks, index_name, timeout=300) -> None:
+    LOGGER.info('waiting for index {} to be built'.format(index_name))
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        result = node.run_nodetool(f"viewbuildstatus {ks}.{index_name}_index",
+                                   ignore_status=True, verbose=False, publish_event=False)
+        if f"{ks}.{index_name}_index has finished building" in result.stdout:
+            LOGGER.info("index {} has been built".format(index_name))
+            return
+        time.sleep(30)
+    raise TimeoutError(f"Timeout error while creating index {index_name}. "
+                       f"stdout\n: {result.stdout}\n"
+                       f"stderr\n: {result.stderr}")
+
+
+def verify_query_by_index_works(session, ks, cf, column) -> None:
+    # get some value from table to use it in query by index
+    result = session.execute(f'SELECT "{column}" FROM {ks}.{cf} limit 1')
+    value = list(result)[0][0]
+    match type(value).__name__:
```

Comment:
Those are all the possible types ?

doesn't user profile or s-b or Gemini might create other types ?

Please consider verifying this nemesis with those as well

Reply:
I tried using more types, found `datetime` needed also quoting.
Generally, number of python types is much more limited than column types and I found only these 3 to require quoting or adding some reformat.

Reply:
you write those 3, and there only two types here in the match/case...

Reply:
seems I didn't push changes... thanks for checking